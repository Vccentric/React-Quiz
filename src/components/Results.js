import React from "react";
import PropTypes from "prop-types";

// results component
const Results = ({
  totalCorrectAnswers,
  totalQuestions,
  resetQuizStats,
  setQuizList,
}) => {
  return (
    <div id="results">
      <h2 className="title">Results:</h2>
      <p className="test-score">
        You have {totalCorrectAnswers} correct answers out of a total of{" "}
        {totalQuestions} questions!!!
      </p>
      <div className="actions-container">
        <button className="action quit" onClick={setQuizList}>
          Quit
        </button>
        <button className="action restart" onClick={resetQuizStats}>
          Restart
        </button>
      </div>
    </div>
  );
};

Results.propTypes = {
  totalCorrectAnswers: PropTypes.number,
  totalQuestions: PropTypes.number,
  resetQuizStats: PropTypes.func,
  setQuizList: PropTypes.func,
};

export default Results;
