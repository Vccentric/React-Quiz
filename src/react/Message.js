import React from 'react';
import { hot } from 'react-hot-loader';

// message component
function Message(props) {
    const { isAnswered, isCorrect } = props;
    return (
        <div
            id="message"
            style={isAnswered ? { display: 'block' } : { display: 'none' }}
            className={isCorrect ? 'correct' : 'wrong'}
        >
            {isCorrect ? 'Correct Answer!' : 'Wrong Answer!'}
        </div>
    );
}

export default hot(module)(Message);
