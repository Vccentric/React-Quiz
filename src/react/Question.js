import React from 'react';
import { hot } from 'react-hot-loader';

// question component
function Question(props) {
    const { question } = props;
    return <p id="question">{question}</p>;
}

export default hot(module)(Question);
