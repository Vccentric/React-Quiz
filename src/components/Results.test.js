import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import React from "react";
import Results from "./Results";

// setup variables
const resetQuizStats = jest.fn();
const setQuizList = jest.fn();

test("renders correctly - snapshot", () => {
  const tree = renderer
    .create(
      <Results
        totalCorrectAnswers={0}
        totalQuestions={10}
        resetQuizStats={resetQuizStats}
        setQuizList={setQuizList}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders results with mock values", () => {
  const { container } = render(
    <Results
      totalCorrectAnswers={1}
      totalQuestions={5}
      resetQuizStats={resetQuizStats}
      setQuizList={setQuizList}
    />
  );
  const element = container.getElementsByClassName("test-score");
  expect(element?.[0]?.textContent).toBe(
    "You have 1 correct answers out of a total of 5 questions!!!"
  );
});
