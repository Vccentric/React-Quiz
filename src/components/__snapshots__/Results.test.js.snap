// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - snapshot 1`] = `
<div
  id="results"
>
  <h2
    className="title"
  >
    Results:
  </h2>
  <p
    className="test-score"
  >
    You have 
    0
     correct answers out of a total of
     
    10
     questions!!!
  </p>
  <div
    className="actions-container"
  >
    <button
      className="action quit"
      onClick={[MockFunction]}
    >
      Quit
    </button>
    <button
      className="action restart"
      onClick={[MockFunction]}
    >
      Restart
    </button>
  </div>
</div>
`;
